jujugui/static/gui/src/app/jujulib/test-charmstore.js

```diff
@@ -498,6 +498,38 @@ describe('jujulib charmstore', function() {
     });
   });
 
+  describe('getCanonicalId', function() {
+    it('makes a request to fetch the canonical id for an entity', function() {
+      const callback = sinon.stub();
+      charmstore.getCanonicalId('cs:xenial/ghost-4', callback);
+      const sendGetRequest = charmstore.bakery.sendGetRequest;
+      assert.equal(sendGetRequest.callCount, 1);
+      assert.equal(
+        sendGetRequest.args[0][0], 'local/v5/xenial/ghost-4/meta/id');
```

Comment:
The following part of the test would be easier to read if we assigned sendGetRequest.args[0][0] and sendGetRequest.args[0][1] to variables, and then assert on those variables.